tests/php/classes/test-class-wc-rest-connect-shipping-label-controller.php

```diff
@@ -0,0 +1,422 @@
+		// Given.
+		$controller = new WC_REST_Connect_Shipping_Label_Controller( $this->api_client_mock, $this->settings_store, $this->connect_logger_mock, $this->shipping_label );
+		$this->set_store_and_settings_to_be_eligible_for_shipping_label_creation();
+		$this->update_shipping_labels_account_settings( false, $this->settings_store );
+		$product = $this->create_simple_product();
+		$order   = WC_Helper_Order::create_order( 1, $product );
+
+		// When.
+		$request = new WP_REST_Request( 'GET', '/wc/v1/connect/label/' . $order->get_id() . '/creation_eligibility' );
+		$request->set_param( 'order_id', $order->get_id() );
+		$response = $controller->get_creation_eligibility( $request );
+
+		// Then.
+		$this->assertEquals( 200, $response->status );
+		$expected_data = array(
+			'is_eligible' => false,
+			'reason'      => 'account_settings_disabled',
+		);
+		$this->assertEquals( $expected_data, $response->get_data() );
+	}
+
+	/**
+	 * Test that a WC store in a supported country is eligible for shipping label creation while fulfilling all the other requirements.
+	 */
+	public function test_store_in_supported_countries_is_eligible_for_shipping_label_creation() {
+		// Given.
+		$controller = new WC_REST_Connect_Shipping_Label_Controller( $this->api_client_mock, $this->settings_store, $this->connect_logger_mock, $this->shipping_label );
+		$this->set_store_and_settings_to_be_eligible_for_shipping_label_creation();
+		$product = $this->create_simple_product();
+		$order   = WC_Helper_Order::create_order( 1, $product );
+
+		// When.
+		$request = new WP_REST_Request( 'GET', '/wc/v1/connect/label/' . $order->get_id() . '/creation_eligibility' );
+		$request->set_param( 'order_id', $order->get_id() );
+		$response = $controller->get_creation_eligibility( $request );
+
+		// Then.
+		$this->assertEquals( 200, $response->status );
+		$expected_data = array(
+			'is_eligible' => true,
+		);
+		$this->assertEquals( $expected_data, $response->get_data() );
+	}
+
+	/**
+	 * When the client cannot create a customs form, test that a WC store outside the US is not eligible for shipping label creation.
+	 */
+	public function test_non_US_store_is_not_eligible_for_shipping_label_creation_when_client_cannot_create_customs_form() {
+		// Given.
+		$controller = new WC_REST_Connect_Shipping_Label_Controller( $this->api_client_mock, $this->settings_store, $this->connect_logger_mock, $this->shipping_label );
+		$this->set_store_and_settings_to_be_eligible_for_shipping_label_creation();
+		update_option( 'woocommerce_default_country', 'PR' );
+		$product = $this->create_simple_product();
+		$order   = WC_Helper_Order::create_order( 1, $product );
+
+		// When.
+		$request = new WP_REST_Request( 'GET', '/wc/v1/connect/label/' . $order->get_id() . '/creation_eligibility' );
+		$request->set_param( 'order_id', $order->get_id() );
+		$request->set_param( 'can_create_customs_form', false );
```

Comment:
Yep, and thanks for the suggestion! mobile client is going to make requests with values like `true`/`false`. In https://github.com/Automattic/woocommerce-services/pull/2334/commits/bc82936913191b72b443394f575dc7b845aeae8e, I updated to pass a string boolean in the test cases (and they failed in the first place) and parse the boolean params with `filter_var( $request[param_name], FILTER_VALIDATE_BOOLEAN )`

Reply:
Nice, Double checked `FILTER_VALIDATE_BOOLEAN` will indeed returns true for "1", "true", "on" and "yes". False otherwise. [ref](https://www.php.net/manual/en/filter.filters.validate.php)